rts/System/Sound/OpenAL/Sound.cpp

```diff
@@ -493,40 +493,58 @@ void CSound::OpenLoopbackDevice(const std::string& deviceName)
 		return;
 	}
 
+	LOG("[Sound::%s] SDL audio device(s): ", __func__);
+	for (int i = 0, n = SDL_GetNumAudioDevices(0); i < n; ++i) {
+		LOG("[Sound::%s]  * \"%d\" \"%s\"", __func__, i, SDL_GetAudioDeviceName(i, 0));
+	}
+
 	SDL_AudioSpec desiredSpec;
 	SDL_AudioSpec obtainedSpec;
 
-	desiredSpec.channels = 2;
 	desiredSpec.format = AUDIO_S16SYS;
 	desiredSpec.freq = 44100;
 	desiredSpec.padding = 0;
 	desiredSpec.samples = 4096;
 	desiredSpec.callback = RenderSDLSamples;
 	desiredSpec.userdata = this;
+	
+	/* SDL bug: can return devices with >2 channels (3D surround), even if we ask for just 2.
+ 	* (if tested) This causes the 2 "primary" channels to be moved in the 3D space compared to their "normal" state.
+ 	* (if not tested) We are afraid that this risks the 2 "primary" channels to be moved in the 3D space compared to their "normal" state, but this remains to be tested.
+ 	* For this reason, the engine rejects such devices down the road. Don't let it get that far and retry instead.
+ 	*
+ 	* Note that proper support for 3D surround sounds sounds hard, for example as of 2023 counter-strike has very weak support
+ 	* for it, apparently noticeably worse than just stereo according to players, despite being in a more relevant genre. */
+	for(int channelsDesired = 2; channelsDesired > 0; channelsDesired--) {
```

Comment:
As a matter of personal preference I'd rewrite it as:
`for(int channelDesired : {2, 1}) {`

Up to you which options you like the most.

Reply:
nice, didn't know that syntax